justfile

```
@@ -0,0 +1,177 @@
+# dependencies like `dbt` need the virtual environment activated first, which is
+# why `poetry run` is so handy. For `pip` and `uv`, we activate the virtual 
+# environment before running these commands in the same shell. You can see
+# the logic and recipes for this at the end of the file in the 'Developer recipes'.
+#
+# Start of user configuration:
+##############################################################################
+venv_name := '.venv'
+pm := "pip"
+##############################################################################
+# End of user configuration.
+##############################################################################
+
+##############################################################################
+# User recipes.
+# These are the codegen commands you can run with `just`.
+# You can add more recipes in this section for your workflow.
+# If you want to share a recipe you create we welcome PRs to the 
+# `dbt-codgen` repo! -> https://github.com/dbt-labs/dbt-codegen
+#
+# Check out the just documentation here for more on writing your own recipes:
+# https://just.systems/
+##############################################################################
+
+# List all recipes.
+default:
+  @just --list --list-prefix=" ✴︎ "
+
+# Generate _sources.yml for all tables in a schema.
+dbt-generate-source database schema:
+    @{{dbt}} run-operation codegen.generate_source --args '{"schema_name": "{{schema}}", "database_name": "{{database}}"}'
+
+alias gso := dbt-generate-source
+
+# Generate a sql model for a table defined in your sources YAML.
+dbt-generate-base-model source table:
+    @{{dbt}} run-operation codegen.generate_base_model --args '{"source_name": "{{source}}", "table_name": "{{table}}"}'  > generated/stg_{{table}}.sql
+    @awk '/with source as \(/{p=1} p' generated/stg_{{table}}.sql > temp && mv temp generated/stg_{{table}}.sql
+    @echo "Model {{table}} generated in generated/stg_{{table}}.sql"
+
+alias gbase := dbt-generate-base-model
+
+# Generate YAML config with all columns from a SQL model file.
+generated_default := 'generated'
+dbt-generate-model-yaml model_name generated_folder=generated_default:
+    @if [ ! -d "{{generated_folder}}" ]; then \
+        mkdir -p {{generated_folder}}; \
+    fi
+    @{{dbt}} run-operation codegen.generate_model_yaml --args '{"model_names": ["{{model_name}}"]}' > /tmp/{{model_name}}.tmpyml
+    @awk '/models:/{p=1} p' /tmp/{{model_name}}.tmpyml > /tmp/temp{{model_name}} && mv /tmp/temp{{model_name}} {{generated_folder}}/{{model_name}}.yml
+    @echo "Model {{model_name}} generated in {{generated_folder}}/{{model_name}}.yml"
+
+alias gmy := dbt-generate-model-yaml
+
+# Generate YAML config for any unconfigured SQL models.
+# Optionally accepts a parameter for folder to search in.
+default_folder := 'models'
+dbt-generate-missing-yaml folder=default_folder:
+    @for sql_file in $(find {{folder}} -type f -name '*.sql'); do \
```

```suggestion
    @for sql_file in $(find {{folder}} -type f -name '\*.sql' ! -name '\*\_v\[0-9\]\*.sql' ! -name '\*\_V\[0-9\]\*.sql'); do \\
```

this should be changed to acomidate the pattern of having versioned sql files with a singular yml definition.